pcd:
    pcd_path: /home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/pcd/ntu_night_08/0422/sorted_lamps.txt
    cluster_num: 134
    use_virtual_center: false
    virtual_center_path: /home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/pcd/ntu_night_08/0422/virtual_centers.txt

prior_pose:
    downsampled_pose_path: /home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/pcd/ntu_night_08/0422/downsampled_poses.txt
    prior_pose_path: /home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/pcd/ntu_night_08/0422/selected_poses.txt

common:
    img_topic: /d455b/color/image_raw/compressed
    box_topic: /yolov7_bbox
    odom_topic: /odom
    imu_topic: /vn100/imu

camera:
    cam_fx: 383.9298163754784
    cam_fy: 382.9589851049001 
    cam_cx: 322.81111219908314 
    cam_cy: 250.9729552053214
    res_x: 640
    res_y: 480
    distortion: [-0.04406471805202435, 0.04043629293187546, -1.2130737779239174e-06, 0.0018435172104962858]
    dist_model: radtan

imu:
    gyro_std: [0.0001, 0.0001, 0.0001]
    accel_std: [0.001, 0.001, 0.001]
    accel_bias_std: [0.0001, 0.0001, 0.0001]
    gyro_bias_std: [0.0001, 0.0001, 0.0001]
    Rci: [-0.0150710417,  0.9998528146,  0.00819834,
          0.005988709,  -0.0081088609,  0.9999491896,
          0.9998684909,  0.0151193734, -0.0058656186]
    pci: [-0.0110886805, -0.1146477232, -0.0561546573]
    Roi: [1, 0, 0,
          0, 1, 0,
          0, 0, 1]
    accel_norm: 1
    gravity_mag: 9.81

odom:
    vel_std: [0.005, 0.005, 0.005]

state:
    which_group: 2 # -2: msckf, -1: no group, 0: features are grouped in IMU state, 1: features are grouped in relative transform,
                   #  2: features are grouped between clone state and relative transform
    max_clone_size: 11
    max_slam_features: 60
    max_slam_in_update: 35
    max_msckf_in_update: 1000

update:
    do_update_msckf: true
    do_update_slam: true
    do_update_map: true
    do_update_odom: true
    do_update_plane: false
    use_match_extension: true
    msckf_iekf_sigma_px: 1.5
    msckf_chi2_multipler: 3
    slam_sigma_px: 1.5
    slam_chi2_multipler: 3
    zupt_chi2_multipler: 1
    map_sigma_px: 1
    map_chi2_multipler_dl: 4
    map_chi2_multipler_bi: 4
    odom_chi2_multipler: 5
    plane_chi2_multipler_loc: 1.0
    plane_chi2_multipler_prior: 10.0
    plane_distance_weight_loc: 0.02
    plane_distance_weight_prior: 0.02
    plane_distance_threshold_loc: 1.5
    plane_distance_threshold_prior: 5.0

match:
    z_th: 55
    update_z_th: 75
    extend: 20
    large_extend: 20
    grey_th: 252
    grey_th_low: 240
    alpha: 0.2
    ang_th_dl: 0.9998
    ang_th_bi: 0.9998
    large_off: 30
    dl_area: 40
    dl_filter: true
    bi_filter: true
    remain_match: true

init:
    viwo_init:
        init_window_time: 1.0
        init_imu_thresh: 1.0
        gravity_mag: 9.81
        init_max_features: 50
    map_init:
        need_init_in_map: false
        preset_Rwi: [0.385696406, 0.922625533, -0.000215897, 0.000601592]
        preset_pwi: [195.735888597, -8.567298343, 7.179705161]

feat_init:
    triangulate_1d: false
    refine_features: true
    max_runs: 5
    init_lamda: 1e-3
    max_lamda: 1e10
    min_dx: 1e-6
    min_dcost: 1e-6
    lam_mult: 10
    min_dist: 0.25
    max_dist: 150
    max_baseline: 200
    max_cond_number: 20000

tracker:
    histogram_method: 1 # 0: NONE, 1: HISTOGRAM, 2: CLAHE
    num_features: 200
    fast_threshold: 10
    grid_x: 32
    grid_y: 18
    min_px_dist: 8
    pyr_levels: 5
    win_size: 15

zupt:
    try_zupt: true
    zupt_max_velocity: 0.2
    zupt_noise_multiplier: 10
    zupt_max_disparity: 0.5
    zupt_max_velocity_odom: 0.05
    zupt_only_at_beginning: false

tracking_recover:
    dist_th: 35
    ang_th: 0.3
    lost_dist_th: 1.5
    lost_ang_th: 1.5
    z_th: 25
    extend: 20
    grey_thresh_low: 230
    expansion_proj: 40
    subscore_weight: 1
    variance: 64
    search_dist_scope: 45
    search_bar_height: 60
    area_th: 150
    expected_weight: 0.2
    outlier_cost: 0.5
    sigma_pix: 1
    chi2_multipler_dl_maylost: 8
    chi2_multipler_bi_maylost: 8
    chi2_multipler_dl_lost: 4
    chi2_multipler_bi_lost: 4
    prior_z_diff_th: 1

save:
    save_total_state: true
    of_state_est: "/home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/log/state_estimation.txt"
    of_state_std: "/home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/log/state_deviation.txt"
    of_state_loc: "/home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/log/state_tum_loc.txt"
    of_state_global: "/home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/log/state_tum_global.txt"
    of_state_tracking_recover: "/home/gtx/ws_Night_Rider_pp/src/Night_Rider_pp/log/"